front/app/components/admin/InviteUsersWithSeatsModal/index.tsx

```diff
@@ -0,0 +1,116 @@
+import Modal from 'components/UI/Modal';
+import SeatInfo, {
+  SeatTypeMessageDescriptor,
+  TSeatType,
+} from 'components/SeatInfo';
+import Error from 'components/UI/Error';
+
+// Translation
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import messages from './messages';
+
+interface InviteUsersWithSeatsModalProps {
+  showModal: boolean;
+  closeModal: () => void;
+  inviteUsers: () => void;
+  noOfSeatsToAdd: number;
+  seatType: TSeatType;
+}
+
+const InviteUsersWithSeatsModal = ({
+  showModal,
+  closeModal,
+  inviteUsers,
+  noOfSeatsToAdd,
+  seatType,
+}: InviteUsersWithSeatsModalProps) => {
+  const { formatMessage } = useIntl();
+  const [hasAcknowledged, setHasAcknowledged] = useState(false);
+  const [showWarning, setShowWarning] = useState(false);
+  const seatTypeMessages: SeatTypeMessageDescriptor = {
+    admin: messages.admin,
+    collaborator: messages.collaborator,
+  };
+  const seatTypesMessages: SeatTypeMessageDescriptor = {
+    admin: messages.admins,
+    collaborator: messages.collaborators,
+  };
+  const seatTypeModalTitles: SeatTypeMessageDescriptor = {
+    admin: messages.giveAdminRights,
+    collaborator: messages.giveCollaboratorRights,
+  };
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.stopPropagation();
+    setShowWarning(hasAcknowledged);
+    setHasAcknowledged(!hasAcknowledged);
+  };
+
+  const handleConfrimClick = () => {
+    if (!hasAcknowledged) {
+      setShowWarning(true);
+      return;
+    }
+    closeModal();
+    inviteUsers();
+  };
+
+  return (
+    <Modal
+      opened={showModal}
+      close={closeModal}
+      header={
+        <Text color="primary" my="8px" fontSize="l" fontWeight="bold" px="2px">
+          {formatMessage(seatTypeModalTitles[seatType])}
+        </Text>
+      }
+    >
+      <Box display="flex" flexDirection="column" width="100%" p="32px">
+        <Text color="textPrimary" fontSize="m" my="0px">
+          <FormattedMessage
+            {...messages.descriptionMessage}
+            values={{
+              noOfUsers: noOfSeatsToAdd,
+              seatType: formatMessage(seatTypeMessages[seatType]),
+            }}
+          />
+        </Text>
+        <Box py="32px">
+          <SeatInfo seatType={seatType} />
+        </Box>
+
+        <Checkbox
+          checked={hasAcknowledged}
+          onChange={onChange}
+          label={
+            <Text color="blue500" my="0px">
+              {formatMessage(messages.acknowledgement, {
+                seatTypes: formatMessage(seatTypesMessages[seatType]),
+              })}
+            </Text>
+          }
+        />
+
+        {showWarning && (
+          <Box mt="12px">
+            <Error text={formatMessage(messages.acceptWarning)} />
+          </Box>
+        )}
+
+        <Box display="flex" mt="32px">
+          <Button disabled={!hasAcknowledged}>
+            <Box onClick={handleConfrimClick} data-testid="confirm-button-text">
```

Comment:
This is a bit odd. `onClick` should be on the button and a disabled button should not be clickable.

Reply:
Yeah, that was because of the requirement. Without it, we should be good